instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -262,13 +288,25 @@ def _set_api_gateway_v2_proxy_attributes(
     return span
 
 
+def _safe_execute_hook(hook_func, *args):
+    if not callable(hook_func):
+        return
+
+    try:
+        hook_func(*args)
+    except Exception:  # pylint: disable=broad-except
+        logger.warning("Failed executing hook")
```

Comment:
Maybe add the exception message to the warning record

Reply:
done